sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -267,6 +267,7 @@ object SQLConf {
       "Note: For structured streaming, this configuration cannot be changed between query " +
       "restarts from the same checkpoint location.")
     .intConf
+    .checkValue(_ > 0, "the value of sql.shuffle.partitions must be greater than 0")
```

Comment:
@WindCanDie, can you address this comment too (https://github.com/apache/spark/pull/23835/files#r257884912)? it should be `spark.sql.shuffle.partitions`